src/components/Card/CardList.test.tsx

```diff
@@ -29,37 +29,37 @@ describe("CardList component", () => {
     jest.clearAllMocks();
   });
 
-  it("renders a list of all cards if no category is supplied", () => {
-    const params = {};
-    jest.spyOn(ReactRouterDOM, "useParams").mockReturnValue(params);
+  it("renders a list of all cards if no category is supplied", async () => {
+    const query = {};
+    (useRouter as jest.Mock).mockReturnValue({ query });
 
-    const { getByRole } = render(<CardList {...mockCardListProps} />);
-    const cardItems = getByRole("list");
+    const { findByRole } = render(<CardList {...mockCardListProps} />);
```

Comment:
Prefer `findByRole` over `getByRole` because it doesn't need `await` or `waitFor`: https://github.com/testing-library/testing-library-docs/pull/1242